test/infrastructure/docker/api/v1alpha3/conversion.go

```diff
@@ -192,11 +175,52 @@ func Convert_v1beta2_DockerMachineSpec_To_v1alpha3_DockerMachineSpec(in *infrav1
 }
 
 func Convert_v1beta2_DockerClusterStatus_To_v1alpha3_DockerClusterStatus(in *infrav1.DockerClusterStatus, out *DockerClusterStatus, s apiconversion.Scope) error {
-	return autoConvert_v1beta2_DockerClusterStatus_To_v1alpha3_DockerClusterStatus(in, out, s)
+	if err := autoConvert_v1beta2_DockerClusterStatus_To_v1alpha3_DockerClusterStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// Reset conditions from autogenerated conversions
+	// NOTE: v1beta2 conditions should not be automatically be converted into v1alpha3 conditions.
+	out.Conditions = nil
+	if in.Deprecated != nil && in.Deprecated.V1Beta1 != nil && in.Deprecated.V1Beta1.Conditions != nil {
+		clusterv1alpha3.Convert_v1beta2_Deprecated_V1Beta1_Conditions_To_v1alpha3_Conditions(&in.Deprecated.V1Beta1.Conditions, &out.Conditions)
+	}
+
+	if in.Initialization != nil {
+		out.Ready = in.Initialization.Provisioned
+	}
+
+	// Move FailureDomains
+	if in.FailureDomains != nil {
+		out.FailureDomains = clusterv1alpha3.FailureDomains{}
+		for _, fd := range in.FailureDomains {
+			out.FailureDomains[fd.Name] = clusterv1alpha3.FailureDomainSpec{
+				ControlPlane: fd.ControlPlane,
+				Attributes:   fd.Attributes,
+			}
+		}
+	}
+
+	return nil
 }
 
 func Convert_v1beta2_DockerMachineStatus_To_v1alpha3_DockerMachineStatus(in *infrav1.DockerMachineStatus, out *DockerMachineStatus, s apiconversion.Scope) error {
-	return autoConvert_v1beta2_DockerMachineStatus_To_v1alpha3_DockerMachineStatus(in, out, s)
+	if err := autoConvert_v1beta2_DockerMachineStatus_To_v1alpha3_DockerMachineStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// Reset conditions from autogenerated conversions
+	// NOTE: v1beta2 conditions should not be automatically be converted into v1alpha3 conditions.
+	out.Conditions = nil
+	if in.Deprecated != nil && in.Deprecated.V1Beta1 != nil && in.Deprecated.V1Beta1.Conditions != nil {
+		clusterv1alpha3.Convert_v1beta2_Deprecated_V1Beta1_Conditions_To_v1alpha3_Conditions(&in.Deprecated.V1Beta1.Conditions, &out.Conditions)
+	}
+
+	if in.Initialization != nil && in.Initialization.Provisioned {
```

Comment:
Let's also set ready if Provisioned is false please. I know it comes down to the same thing, but it's just simpler

Reply:
We cannot distinguish the false is nil or actual false since the type isn't pointer.

```
                -               Initialization:         &v1beta2.DockerMachineInitializationStatus{Provisioned: &false},
                +               Initialization:         nil,
```

So it seems to be good to set Provisioned only if the value is true